### PR TITLE
Add boot options to vsphere_guest module.

### PR DESCRIPTION
**Pull request by silid**
##### SUMMARY
vsphere_guest allows the creation and reconfiguration of virtual machine, however doesn't currently permit the setting of boot options, ie. boot device order and boot delay.

This modification allows the setting of those values.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
vsphere_guest

##### ANSIBLE VERSION

```
ansible 2.4.0
  config file = None
  configured module search path = [u'/home/silid/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]

```

##### ADDITIONAL INFORMATION
When setting the boot order for disks and ethernet you need to know the device key of the disk and nic you want to add.

### CODE REVIEW

**Comment by silid** [comment]
ready_for_review

**Comment by silid** [comment]
@jctanner

File: lib/ansible/modules/cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -875,6 +895,40 @@ def reconfigure_vm(vsphere_client, vm, module, esxi, resource_pool, cluster_name
         spec.set_element_extraConfig(extra_config)
         changes["extra_config"] = vm_extra_config
 
+    # Change boot options
+    if vm_boot_options:
+        spec = spec_singleton(spec, request, vm)
+        boot_options = VI.ns0.VirtualMachineBootOptions_Def('boot_options').pyclass()
+        if 'boot_delay' in vm_boot_options:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vsphere_guest.py:902]: Could you please move this to a common function?

**Review by Akasurde — CHANGES_REQUESTED** [review]
Rest LGTM

**Comment by Akasurde** [comment]
@silid Any news ?

**Comment by silid** [comment]
It seems vsphere_guest was removed in 2.5 so this pull request is no longer relevant.

**Pull request closed without merging**